include/nbl/builtin/hlsl/scan/parameters_struct.hlsl

```diff
@@ -0,0 +1,30 @@
+#ifndef _NBL_HLSL_SCAN_PARAMETERS_STRUCT_INCLUDED_
+#define _NBL_HLSL_SCAN_PARAMETERS_STRUCT_INCLUDED_
+
+#define NBL_BUILTIN_MAX_SCAN_LEVELS 7
+
+#ifdef __cplusplus
+#define uint uint32_t
+#endif
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace scan
+{
+	// REVIEW: Putting topLevel second allows better alignment for packing of constant variables, assuming lastElement has length 4. (https://learn.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-packing-rules)
+	struct Parameters_t {
```

Comment:
we will now use `scalar_layout` on ALL SSBO and UBOs, just remember to enable it in device features when creating the device

@Erfan-Ahmadi @deprilula28 is it already the default compilation option with DXC?

Reply:
They're not the default, we'd need to use -fvk-use-scalar-layout https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/SPIR-V.rst#memory-layout-rules

Should that be enabled always?

Reply:
Yes, we should enable that always by default. I'll make sure to make this change in the Nabla repo for cad works. Before we sync again with n4ce

Reply:
done: https://github.com/Devsh-Graphics-Programming/Nabla/commit/8f059418992fff7dffe59af82fc4502d26b072c9